addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PromotionHelper.java

```diff
@@ -117,10 +117,6 @@ public void updatePathPromoteMetrics( int total, PathsPromoteResult result )
         for ( final String path : paths )
         {
             final Transfer transfer = downloadManager.getStorageReference( source, path );
-            if ( transfer == null || !transfer.exists() )
```

Comment:
So does this mean we do not care about if the the transfer exists for the source?

Reply:
We need. I just moved the check to PromotionManager where we can set a proper error or skipped flag.

Reply:
In fact, not move. It is already there. This is just a dup check, and causes the promotion to fail too early.